hostwatch/status_syncer.go

```diff
@@ -0,0 +1,67 @@
+package hostwatch
+
+import (
+	"fmt"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/rancher/go-rancher-metadata/metadata"
+	"github.com/rancher/kubernetes-agent/kubernetesclient"
+)
+
+type hostStatusSyncer struct {
+	kClient        *kubernetesclient.Client
+	metadataClient metadata.Client
+}
+
+var (
+	maxRetryCount int = 3
+)
+
+func (h *hostStatusSyncer) syncHoststatus(version string) {
+	err := statusSync(h.kClient, h.metadataClient)
+	if err != nil {
+		log.Errorf("Error syncing host status: [%v]", err)
+	}
+}
+
+func statusSync(kClient *kubernetesclient.Client, metadataClient metadata.Client) error {
+	hosts, err := metadataClient.GetHosts()
+	if err != nil {
+		return fmt.Errorf("Error reading host list from metadata service: [%v]", err)
+	}
+	for _, host := range hosts {
+		mNodeStatus := host.State
+		switch mNodeStatus {
+		case ActiveState:
+			err = cordonUncordon(host, kClient, metadataClient, false)
+		case InActiveState:
+			err = cordonUncordon(host, kClient, metadataClient, true)
+		case EvictedState:
```

Comment:
lets remove eviction stuff from this PR for now. You can save it on your local branch